kernel_gateway/services/kernels/handlers.py

```diff
@@ -19,6 +20,7 @@ class MainKernelHandler(TokenAuthorizationMixin,
     """Extends the notebook main kernel handler with token auth, CORS, and
     JSON errors.
     """
+    @gen.coroutine
     def post(self):
```

Comment:
In order for this to work, you need to wait on the Future returned by `super().post`. Since that's the only async thing here, you can either:

1. don't wrap in `gen.coroutine` and `return super().post(...)`
2. wrap in `gen.coroutine` and `yield super()...`